### PR TITLE
M2P-537 Add support for Magento native In-Store Pickup step 4

### PR DESCRIPTION
**Pull request by dumega**
# Description

As we decide to add support BOPIS Spilt shipping to M2, and In-store delivery method is supported with version 2.4.0 of Magento (https://devdocs.magento.com/guides/v2.4/inventory/release-notes.html#v120) , this PR is for adding support for this feature.

Step 4 : Create order with in-store pickup option.

Fixes: https://boltpay.atlassian.net/browse/M2P-537

#changelog Add support for Magento native In-Store Pickup step 4

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [x] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [ ] Successfully tested locally (or docker image)
- [ ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# For PR Reviewer 
- [ ] Reviewed unit tests to make sure we are using real components rather than mocks as much as possible?
- [ ] For any major change (observer, new Bolt feature, core Magento interaction) we must add a feature switch, did you verify this?

# Checklist:

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have created or modified unit tests to sufficiently cover my changes.
- [ ] I have added my Jira ticket link and provided a changelog message.

### CODE REVIEW

**Review by danac-gs — APPROVED** [review]

File: Helper/Order.php [hunk 1]
```diff
@@ -511,14 +513,21 @@ protected function setAddress($quoteAddress, $address)
...
-        $address = $transaction->order->cart->shipments[0]->shipping_address ?? null;
-        $referenceShipmentMethod = $transaction->order->cart->shipments[0]->reference ?? null;
-
-        if ($address) {
-            $this->setAddress($quote->getShippingAddress(), $address);
-            if (isset($referenceShipmentMethod) && $this->configHelper->isPickupInStoreShippingMethodCode($referenceShipmentMethod)) {
-                $addressData = $this->configHelper->getPickupAddressData();
-                $quote->getShippingAddress()->addData($addressData);
+        if (isset($transaction->order->cart->in_store_shipments)) {
+            $address = $transaction->order->cart->in_store_shipments[0]->shipment->shipping_address ?? null;
```
Comment (vitaliyreznikov) [inline Helper/Order.php:517]: can we add check if `$transaction->order->cart->in_store_shipments[0]->shipment->shipping_address` set?

File: ThirdPartyModules/Magento/InStorePickupShipping.php [hunk 2]
```diff
@@ -250,6 +250,67 @@ public function getShipToStoreCarrierMethodCodes(
...
+     * @return array
+     */
+    public function setInStoreShippingAddressForPrepareQuote(
+        $addressConverter,
+        $getPickupLocation,
+        $quote,
+        $transaction
+    ) {
+        try {
+            $shipment = $transaction->order->cart->in_store_shipments[0]->shipment;
```
Comment (vitaliyreznikov) [inline ThirdPartyModules/Magento/InStorePickupShipping.php:293]: can we add isset here?

**Review by vitaliyreznikov — APPROVED** [review]

File: Helper/Order.php (hunk 1 above)
Comment (dumega) [reply Helper/Order.php:517]: Done

File: ThirdPartyModules/Magento/InStorePickupShipping.php (hunk 2 above)
Comment (dumega) [reply ThirdPartyModules/Magento/InStorePickupShipping.php:293]: We already check there https://github.com/BoltApp/bolt-magento2/pull/1208/files#diff-7e87dc817d298f1174413fe57820e176d94a88eed33d0d47ea63c974c6d48fb3R516

**Pull request merged**